### PR TITLE
perf: Further refinements to pathology CLI calls

### PR DESCRIPTION
**Pull request by aauker**
Brings a whole-slide ETL down to a few minutes:

1. Generate tiles, 20x, 128px tiles (150k) - 23 sec
2. Detect tissue using otsu's method - 1 min 50 sec
3. Run a Resnet18 inference classifier - 21 sec
4. Generate 6 PNG markup images - 3 min 4 sec

Example notebook showing functionality: https://mskcc.box.com/s/xwzzoqoompfymkcf34cqvb9wexxvdjs3

### CODE REVIEW

File: pyluna-pathology/luna/pathology/cli/generate_mask.py [hunk 1]
```diff
@@ -52,7 +52,7 @@ def generate_mask(input_slide_image, input_slide_roi, output_dir, annotation_nam
 
     Args:
         input_slide_image (str): path to slide image (.svs)
```
Comment (doori) [inline pyluna-pathology/luna/pathology/cli/generate_mask.py:54]: Do we only support (.svs)? should we say (openslide compatible file format) for all input_slide_image args

File: pyluna-pathology/luna/pathology/cli/generate_tiles.py [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+              help="batch size used for inference speedup", default=64)    
+@click.option('-m', '--method_param_path', required=False,
+              help='path to a metadata json/yaml file with method parameters to reproduce results')
+def cli(**cli_kwargs):
+    """Rasterize a slide into smaller tiles
+    
+    Tiles are saved in the whole-slide tiles binary format (tiles.pil), and the corresponding manifest/header file (tiles.csv) is also generated
+
+    Neccessary data for the manifest file are: 
+    address, x_coord, y_coord, full_resolution_tile_size, tile_image_binary, tile_image_length, tile_image_size_xy, and tile_image_size_mode
```
Comment (doori) [inline pyluna-pathology/luna/pathology/cli/generate_tiles.py:34]: tile_image_size_mode -> tile_image_mode, for all docstrings that have tile_image_size_mode

File: pyluna-pathology/luna/pathology/cli/run_tissue_detection.py [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+            logger.info(f"Starting stain thresholding, channel=1, threshold={threshold_stain1}")
+            df['stain1_score'] = list (tqdm(p.map(partial(compute_stain_score, vectors=stain_vectors, channel=1, stain_threshold=threshold_stain1), [row for _, row in df.iterrows()]), total=len(df)))
+        
+    logger.info (f"Filtering based on query: {filter_query}")
+    df = df.query(filter_query)
+
+    logger.info (df)
+
+    slide_name = Path(input_slide_image).stem
+    output_header_file = f"{output_dir}/{slide_name}-otsu_detections={threshold}.tiles.pil"
```
Comment (doori) [inline pyluna-pathology/luna/pathology/cli/run_tissue_detection.py:173]: should this be a .csv?

File: pyluna-pathology/luna/pathology/cli/infer_tile_labels.py [hunk 4]
```diff
@@ -70,16 +70,18 @@ def predict(self, input_tiles):
         return self.model(input_tiles)
 
 def infer_tile_labels(input_slide_tiles, output_dir, repo_name, transform_name, model_name, weight_tag, num_cores, batch_size):
```
Comment (doori) [inline pyluna-pathology/luna/pathology/cli/infer_tile_labels.py:72]: Can users pass in a path to their trained model instead of a github repo?

**Review by doori — APPROVED** [review]
Nice! couple considerations:
- update entrypoints in pyluna-pathology/setup.cfg with the new CLIs and deprecate generate_tile_labels/visualize_tile_labels

- if the whole slide without tissue detection is not used, would it make sense to save the tiles addresses only in generate_tiles, then save the image binary in run_tissue_detection?
The entire tile images binary at where rgm=scanned mag looks much larger than the WSI (WSI 854M v. tiles.pil 6.9G)

**Comment by aauker** [comment]
Thanks for the careful review Doori, I’ll implement the suggested changes. There’s definitely a more efficient way to generate/detect tiles. As for model loading, yes one can use local code, but there’s a small bit of updating needed after #268

**Pull request merged**